include/embedded_scheduler/scheduler.hpp

```diff
@@ -301,32 +301,41 @@ class EventLoop final
     template <typename ExecutionMonitor = std::monostate>
     [[nodiscard]] SpinResult<Clock> spin()
     {
-        auto next  = time_point::max();
-        auto now   = Clock::now();
-        auto worst = duration::zero();
+        SpinResult<Clock> result{.next_deadline  = time_point::max(),
+                                 .worst_lateness = duration::zero(),
+                                 .approx_now     = time_point::min()};
+        if (tree_.empty())
+        {
+            result.approx_now = Clock::now();
```

Comment:
In original implementation, we always had 2 consecutive `now()` calls on the very first loop iteration for the future  scheduled event. Now we do only minimum (1) call per iteration.

Reply:
Please tag this branch with `[[unlikely]]`. Maybe there are other places that could use this.